maestro-cli/src/main/java/maestro/cli/device/DeviceService.kt

```diff
@@ -0,0 +1,164 @@
+package maestro.cli.device
+
+import com.github.michaelbull.result.expect
+import dadb.Dadb
+import io.grpc.ManagedChannelBuilder
+import ios.idb.IdbIOSDevice
+import maestro.MaestroTimer
+import maestro.cli.CliError
+import maestro.cli.device.ios.Simctl
+import maestro.cli.util.EnvUtils
+import java.io.File
+
+object DeviceService {
+
+    fun startDevice(device: Device): Device {
+        when (device.platform) {
+            Device.Platform.IOS -> {
+                Simctl.launchSimulator(device.id)
+                Simctl.awaitLaunch(device.id)
+                return device
+            }
+            Device.Platform.ANDROID -> {
+                val androidHome = EnvUtils.androidHome()
+                    ?: throw CliError("ANDROID_HOME environment variable is not set. Unable to find 'ANDROID_HOME/tools/emulator' executable.")
```

Comment:
Even if ANDROID_HOME is set, that doesn't not mean the emulator is installed necessarily. We should also check for the existence of the emulator binar